src/woo_publications/metadata/tests/test_load_information_categories.py

```diff
@@ -0,0 +1,100 @@
+
+from ..models import InformationCategory
+from .factories import InformationCategoryFactory
+
+information_categories_fixture = Path(
+    settings.DJANGO_PROJECT_DIR
+    / "metadata"
+    / "tests"
+    / "information_categories_fixture.json",
+)
+
+
+class LoadInformationCategoriesCommandTests(TestCase):
+    def test_load_ic_with_empty_db(self):
+        assert not InformationCategory.objects.exists()
+
+        call_command(
+            "load_information_categories",
+            information_categories_fixture,
+            stdout=StringIO(),
+        )
+
+        self.assertEqual(InformationCategory.objects.count(), 10)
+
+    def test_load_ic_with_random_ics(self):
+        assert not InformationCategory.objects.exists()
+
+        InformationCategoryFactory.create_batch(3)
+
+        call_command(
+            "load_information_categories",
+            information_categories_fixture,
+            stdout=StringIO(),
+        )
+
+        self.assertEqual(InformationCategory.objects.count(), 13)
+
+    def test_load_updated_none_fixture_ic_fields_stay_the_same(self):
+        assert not InformationCategory.objects.exists()
+
+        ic = InformationCategoryFactory.create(
+            order=1010,
+            uuid="be4e21c2-0be5-4616-945e-1f101b0c0e6d",
+            identifier="https://identifier.overheid.nl/tooi/def/thes/kern/c_139c6280",
+            bron_bewaartermijn="bewaartermijn",
+            selectiecategorie="selectiecategorie",
+            archiefnominatie=ArchiveNominationChoices.retain,
+            bewaartermijn=5,
+            toelichting_bewaartermijn="toelichting",
+            omschrijving="omschrijving",
+        )
+        ic2 = InformationCategoryFactory.create(
+            order=1020,
+            uuid="8f3bdef0-a926-4f67-b1f2-94c583c462ce",
+            identifier="https://identifier.overheid.nl/tooi/def/thes/kern/c_aab6bfc7",
+            bron_bewaartermijn="bewaartermijn",
+            selectiecategorie="selectiecategorie",
+            archiefnominatie=ArchiveNominationChoices.retain,
+            bewaartermijn=5,
+            toelichting_bewaartermijn="toelichting",
+            omschrijving="omschrijving",
+        )
+
+        call_command(
+            "load_information_categories",
+            information_categories_fixture,
+            stdout=StringIO(),
+        )
+
+        ic.refresh_from_db()
+        ic2.refresh_from_db()
+
+        self.assertEqual(InformationCategory.objects.count(), 10)
+
+        for obj in [ic, ic2]:
+            self.assertEqual(obj.bron_bewaartermijn, "bewaartermijn")
+            self.assertEqual(obj.selectiecategorie, "selectiecategorie")
+            self.assertEqual(obj.archiefnominatie, ArchiveNominationChoices.retain)
+            self.assertEqual(obj.bewaartermijn, 5)
+            self.assertEqual(obj.toelichting_bewaartermijn, "toelichting")
+            self.assertEqual(obj.omschrijving, "omschrijving")
```

Comment:
Isn't it clearer to setup this test as
1. call command
2. make changes
3. call command again
4. assert changes have been kept?

Or am I misunderstanding the testcase?

Reply:
You are misunderstanding the testcase. I want to test that if some existing data would not overwrite the user input data. So the test is:

1. Create data which correspond in the fixture.
2. call the command.
3. assert that the user data didn't change.